bundles/core/src/main/java/com/adobe/cq/commerce/core/search/internal/models/SearchOptionsImpl.java

```diff
@@ -50,6 +50,24 @@ public SearchOptionsImpl() {
         attributeFilters = new HashMap<>();
     }
 
+    public SearchOptionsImpl(SearchOptions searchOptions) {
+        attributeFilters = searchOptions.getAttributeFilters();
```

Comment:
This should probably be `SearchOptions#getAllFilters()`? 

For the implementation of `getAttributeFilters()` the returned map does not contain the category specific fields.

Reply:
fixed in the lines below